core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/CoreDocumentSynchronizationConfig.java

```diff
@@ -417,6 +417,48 @@ public long getLastKnownHash() {
             break;
         }
         break;
+      case UPDATE:
+        switch (newestChangeEvent.getOperationType()) {
+          case UPDATE:
+            return new ChangeEvent<>(
+                newestChangeEvent.getId(),
+                OperationType.UPDATE,
+                newestChangeEvent.getFullDocument(),
+                newestChangeEvent.getNamespace(),
+                newestChangeEvent.getDocumentKey(),
+                lastUncommittedChangeEvent.getUpdateDescription() != null ?
+                    lastUncommittedChangeEvent
+                        .getUpdateDescription()
+                        .merge(newestChangeEvent.getUpdateDescription()) :
+                    newestChangeEvent.getUpdateDescription(),
+                newestChangeEvent.hasUncommittedWrites()
+            );
+          case REPLACE:
+            return new ChangeEvent<>(
+                newestChangeEvent.getId(),
+                OperationType.REPLACE,
+                newestChangeEvent.getFullDocument(),
+                newestChangeEvent.getNamespace(),
+                newestChangeEvent.getDocumentKey(),
+                null,
+                newestChangeEvent.hasUncommittedWrites()
+            );
+        }
+      case REPLACE:
+        switch (newestChangeEvent.getOperationType()) {
+          case UPDATE:
+            return new ChangeEvent<>(
+                newestChangeEvent.getId(),
+                OperationType.REPLACE,
+                newestChangeEvent.getFullDocument(),
```

Comment:
[q] does a local update always have the full document in the change event?

Reply:
Yes, I cannot think of a case where we do not have the full document.